### PR TITLE
fix documentation for $HOME-windows

### PR DESCRIPTION
**Pull request by mattn**
* using %USERPROFILE% if %HOMEDRIVE% %HOMEPATH% is not set.
* Not passed into external programs

### CODE REVIEW

File: runtime/doc/options.txt [hunk 1]
```diff
@@ -422,8 +422,13 @@ command, not when assigning a value to an option with ":let".
 
 							*$HOME-windows*
 On MS-Windows, if $HOME is not defined as an environment variable, then
-at runtime Vim will set it to the expansion of $HOMEDRIVE$HOMEPATH.
-
+at runtime Vim will set it to the expansion of $HOMEDRIVE$HOMEPATH. If
+$HOMEDRIVE and $HOMEPATH is not set, $USERPROFILE is used. This is an internal
```
Comment (ZyX-I) [inline runtime/doc/options.txt]: The test

```
 +    UnletEnv $HOME
 +    let $USERPROFILE = 'C:\foo'
 +    let $HOMEDRIVE = 'C:'
 +    let $HOMEPATH = '\baz'
 +    call CheckHome('C:\foo')
```

checks that `$USERPROFILE` has priority over `$HOMEDRIVE$HOMEPATH`. Though given that first two lines of CheckHome were commented I can’t say whether this is true based on the test.

Also test

```
 +    UnletEnv $HOME $USERPROFILE $HOMEPATH
 +    let $HOMEDRIVE = 'C:'
 +    call CheckHome('C:\')
```

checks that `$HOMEDRIVE` is enough to get `$HOME`, so “if $HOMEDRIVE and $HOMEPATH is not set” is not true (and also has a typo: should be “are”), need “if $HOMEDRIVE is not set”. Same note about tests not checking anything due to commented out lines applies.

File: runtime/doc/options.txt [hunk 2]
```diff
@@ -422,8 +422,13 @@ command, not when assigning a value to an option with ":let".
...
+value. So the value is not passed into external programs. For example, below
+should return empty result: >
+		:echo system('set | findstr ^HOME=')
+<
```
Comment (ZyX-I) [inline runtime/doc/options.txt]: `<` is only needed if you need whitespace character in the first column. You have `B` there which is not a whitespace, so `<` is not needed.

**Comment by ZyX-I** [comment]
I would spell that as

> On Windows, if $HOME is not present in environment, `$HOME` expands to either `$USERPROFILE` or `$HOMEDRIVE.(empty($HOMEPATH)?'\':$HOMEPATH)`. The expansion value is not exported to the environment:
>
>     :echo system('set | findstr ^HOME=')
>
> or
>
>     :echo luaeval('os.getenv("HOME")')
>
> should echo nothing (an empty string) despite `exists('$HOME')` being true. But issuing `let $HOME = '…'` will create an actual environment variable, exported to the subprocesses.

**Comment by brammool** [comment]
I'll include it.

**Pull request closed without merging**